lib/round-tracker.js

```diff
@@ -0,0 +1,78 @@
+import { createMeridianContract } from './meridian/contract.js'
+
+export async function createRoundGetter (pgClient) {
+  const contract = await createMeridianContract()
+
+  let sparkRound
+  const updateSparkRound = async (meridianRound) => {
+    sparkRound = await mapCurrentMeridianRoundToSparkRound({
+      meridianContractAddress: contract.address,
+      meridianRound: BigInt(meridianRound),
+      pgClient
+    })
+    console.log('SPARK round started: %s', sparkRound)
+  }
+
+  contract.on('RoundStart', updateSparkRound)
+  updateSparkRound(await contract.currentRoundIndex())
+
+  return () => sparkRound
+}
+
+export async function mapCurrentMeridianRoundToSparkRound ({
+  meridianContractAddress,
+  meridianRound,
+  pgClient
+}) {
+  let sparkRound
+
+  const { rows: [lastMeridianRecord] } = await pgClient.query(
+    'SELECT * FROM meridian_contract_versions ORDER BY last_spark_round DESC LIMIT 1'
+  )
+
+  // More events coming from the same meridian contract
+  if (lastMeridianRecord && lastMeridianRecord.contract_address === meridianContractAddress) {
+    sparkRound = BigInt(lastMeridianRecord.spark_round_offset) + meridianRound
```

Comment:
```suggestion
    sparkRoundIndex = BigInt(lastMeridianRecord.spark_round_offset) + meridianRound
```

What do you think about making clear we are talking about the number? Could be id, index, number?

Reply:
I am okay with that. Let's use "number" because SPARK rounds start from the number 1. The value `0` is reserved for existing measurements not connected to any round yet.

Reply:
Yes I think so